bindings_ffi/src/mls.rs

```diff
@@ -917,19 +917,30 @@ impl FfiConversations {
 
     pub async fn sync_all_conversations(&self) -> Result<u32, GenericError> {
         let inner = self.inner_client.as_ref();
+        let conn = inner.store().conn()?;
         let groups = inner.find_groups(GroupQueryArgs::default().include_sync_groups())?;
 
         log::info!(
-            "groups for client inbox id {:?}: {:?}",
+            "Groups for client inbox ID {:?}: {:?}",
             self.inner_client.inbox_id(),
             groups.len()
         );
 
-        let num_groups_synced: usize = inner.sync_all_groups(groups).await?;
-        // Uniffi does not work with usize, so we need to convert to u32
+        // Call sync operations in parallel
+        let (welcome_result, group_result) = tokio::join!(
+            self.inner_client.sync_welcomes(&conn),
+            inner.sync_all_groups(groups)
+        );
```

Comment:
If we do these in parallel, is there a chance that we don't sync the messages on a new group?

Reply:
O good call out!